src/ride_type.h

```diff
@@ -75,7 +76,7 @@ class RideType {
 	virtual RideInstance *CreateInstance() const = 0;
 	void SetupStrings(TextData *text, StringID base, StringID start, StringID end, StringID name, StringID desc);
 
-	const RideTypeKind kind; ///< Kind of ride type.
+	RideTypeKind kind;       ///< Kind of ride type.
```

Comment:
You can make it `const` if you initialize it before entering the constructor, ie

`RideType(RideTypeKind rtk) : kind(rtk) { ... }`

Reply:
The information whether we have a gentle or thrill ride only becomes available during `Load()`, so we still need the possibility to change the `kind` after the ctor.